src/ui/component/copyToClipboard/copyToClipboard.tsx

```diff
@@ -1,32 +1,59 @@
+import { useCallback, useState } from 'react'
 import { useAppStore } from '@/ui/store/appStore'
-import { useCallback } from 'react'
+import { useDebounce } from '@/ui/hook/useDebounce'
 import { Icon } from '@/ui/component/icon/icon'
+import classNames from 'classnames'
 import './copyToClipboard.scss'
 
+type CopyState = 'copy' | 'success' | 'error'
+
 type CopyToClipBoardProps = {
   textToCopy: string
+  animationDelay?: number
   onCopied?: (isCopied: boolean) => void
 }
 
-export const CopyToClipboard = ({ onCopied, textToCopy }: CopyToClipBoardProps): JSX.Element => {
+export const CopyToClipboard = ({
+  animationDelay = 2000,
+  onCopied,
+  textToCopy
+}: CopyToClipBoardProps): JSX.Element => {
   const theme = useAppStore(state => state.theme)
+  const [copyState, setCopyState] = useState<CopyState>('copy')
+
+  const resetCopyState = useCallback((): void => {
+    setCopyState('copy')
+  }, [])
+
+  useDebounce(resetCopyState, copyState !== 'copy' ? animationDelay : null, [copyState])
```

Comment:
When component renders, it executes useDebounce even if we haven't click on an icon, I think.

Reply:
I've removed useDebounce as the hook isn't really meant for this use case. I haven't found a way to prevent the set function to not be called when using the hook though. Searching online the hooks I found all have the hook set and cleared on first render.